plugins/inputs/execd/shim/goshim.go

```diff
@@ -274,6 +282,30 @@ func LoadConfig(filePath *string) ([]telegraf.Input, error) {
 	return loadedInputs, err
 }
 
+func parseConfig(contents []byte, conf interface{}) (toml.MetaData, error) {
```

Comment:
I know this is a little different from what we have in config.go, but let's have only the env var replacement in a new function  `expandEnvVars([]byte)`.  After calling it we can call toml.Decode as it is now.

Reply:
@danielnelson good catch!

https://github.com/influxdata/telegraf/compare/c38465e0c4512b7f3e12ef9e0b1532cf4f813bf4..d4991289c7504d3d85df7cfc718c9052038513c2